### PR TITLE
bug fixes, change permissions for the controller and added validation…

### LINKED ISSUE #8: controller is missing permissions to watch capps
A marker is missing in the code to allow list and watch on `Capps`. This results in errors when deploying the controller, which make the controller restart.

### PR DESCRIPTION
**Pull request by sahar2339**
… for hostname

This commit fixes bug related to kustomize comments and indentation, removed mutating webhook, added permissions to watch capp and validation for hostname

fixes #8 #9 #10 #12 #13

### CODE REVIEW

**Comment by mzeevi** [comment]
The webhooks are not registed in `main.go`.

Also we should probably have a discussion about how we want to the webhook comments to look like. Not a big fan of a separate webhooks messages file since it causes limited webhook messages on deny; changed it on the HNS project as well.

**Comment by sahar2339** [comment]
Pushed new commit, registered the webhook, and fixed typeMeta bugs.
This commit also contains @omerbd21 feature that added Support for pullSecrets.

File: main.go [hunk 1]
```diff
@@ -118,6 +119,9 @@ func main() {
 		setupLog.Error(err, "unable to create controller", "controller", "Capp")
 		os.Exit(1)
 	}
+
+	mgr.GetWebhookServer().Register(wh.ServingPath, &webhook.Admission{Handler: &wh.CappValidator{}})
+
 	//+kubebuilder:scaffold:builder
 
 	// mgr.GetWebhookServer().Register(wh.ServingPath, &webhook.Admission{Handler: &wh.CappValidator{}})
```
Comment (mzeevi) [inline main.go:127]: duplicate code

**Pull request merged**